src/lib/common/sol-platform-impl-linux-micro.c

```diff
@@ -514,16 +522,137 @@ teardown_pid1(void)
     reboot(reboot_cmd);
 }
 
+static bool
+sol_platform_linux_micro_should_debug(struct gdbserver *gdb)
+{
+    struct sol_file_reader *fr;
+    struct sol_str_slice file;
+    struct sol_vector tokens;
+    struct sol_str_slice *token;
+    uint16_t i;
+    bool res = false;
+
+    fr = sol_file_reader_open("/proc/cmdline");
+    if (!fr) {
+        SOL_ERR("Could not open /proc/cmdline");
+        return false;
+    }
+
+    file = sol_file_reader_get_all(fr);
+    // remove the \n in the end of the file (this is a single line file - always)
+    file.len--;
+    tokens = sol_util_str_split(file, " ", 0);
+    SOL_VECTOR_FOREACH_IDX (&tokens, token, i) {
+        char arg[PATH_MAX] = { 0 };
+        int err;
+
+        err = snprintf(arg, sizeof(arg), "%.*s", SOL_STR_SLICE_PRINT(*token));
+        if (err < 0 || err >= (int)sizeof(arg)) {
+            SOL_ERR("Could not format arg");
+            continue;
+        }
+
+        if (streq(arg, SOL_DEBUG_ARG)) {
+            res = true;
+        } else if (strstartswith(arg, SOL_DEBUG_COMM_ARG)) {
+            gdb->comm = strdup(arg);
+            if (res) {
+                SOL_DBG("Got "SOL_DEBUG_ARG " flag and comm: %s", arg);
+                break;
+            }
+        }
+    }
+
+    // if no comm is provided set the default one
+    if (!gdb->comm) {
+        gdb->comm = strdup("/dev/tty0");
+        if (!gdb->comm) {
+            SOL_INF("Could not allocate gdbserver comm desc memory");
+            return false;
+        }
+        SOL_INF("No debug comm provided, using default one (/dev/tty0)");
+    }
+
+    sol_file_reader_close(fr);
+    return res;
+}
+
+static void
+on_fork(void *data)
+{
+    int err;
+    unsigned int i;
+    char pids[PATH_MAX];
+    struct gdbserver *gdb = data;
+
+    const char *paths[] = {
+        "/usr/bin/gdbserver",
+        "/bin/gdbserver",
+    };
+
+    err = snprintf(pids, sizeof(pids), "%d", gdb->pid);
+    if (err < 0 || err >= (int)sizeof(pids)) {
+        SOL_ERR("Could not format pid string to attach gdbserver");
+        sol_platform_linux_fork_run_exit(EXIT_FAILURE);
+    }
+
+    if (setenv("SOL_LOAD_INITIAL_SERVICES", "1", 1) == -1) {
```

Comment:
Yes,  I noticed that as well,  like above this was included just an answer to your review but I'm sure it's useless at this point.  I see also a need to a SOL_PID1_DEBUG or sort of to avoid the pid1 setup at that point.